### ISSUE TITLE
Haddock syntax woes.

### ISSUE THREAD

**Issue by phadej** [emilypi/smash#3]
You have

```haskell
-- 'Smash a b' or 'Maybe (Either a b)'
```

which should be written as

```haskell
-- @'Smash' a b@ or @'Maybe' ('Either' a b)@
```

i.e. at marks code, and quotes delimit an identifier to be linked. If you don't want links, you can use just

```haskell
-- @Smash a b@ or @Maybe (Either a b)@
```

The mistake seems to be systematic, so I won't fix it myself.

**Comment by phadej** [comment]
Kind of relatedly, haddoc linking of operators is tricky. You have write something like:

```
-- @(',')@
```

to link one. E.g. `lens` has

```
src/Control/Lens/Lens.hs:-- @('<<%=') :: 'MonadState' s m => 'LensLike' ((,)a) s s a b -> (a -> b) -> m a@
```

**Comment by emilypi** [comment]
"Woes" is the correct word. I can never get these damned docs right

**Comment by emilypi** [comment]
Closed in #18
